src/wallet/signer.ts

```diff
@@ -0,0 +1,219 @@
+      throw new ValidationError(
+        "For multisigning the transaction must include the SigningPubKey field as an empty string."
+      );
+    }
+
+    if (tx.Signers === undefined) {
+      throw new ValidationError(
+        "For multisigning the transaction must include a Signers field containing an array of signatures."
+      );
+    }
+  });
+
+  const encodedTransactions: string[] = transactions.map((txOrBlob) =>
+    getEncodedTransaction(txOrBlob)
+  );
+
+  return combine(encodedTransactions);
+}
+
+/**
+ * AuthorizeChannel creates a signature that can be used to redeem a specific amount of XRP from a payment channel.
+ *
+ * @param wallet - The account that will sign for this payment channel.
+ * @param channelId - An id for the payment channel to redeem XRP from.
+ * @param amount - The amount in drops to redeem.
+ * @returns A signature that can be used to redeem a specific amount of XRP from a payment channel.
+ */
+function authorizeChannel(
+  wallet: Wallet,
+  channelId: string,
+  amount: string
+): string {
+  const signingData = encodeForSigningClaim({
+    channel: channelId,
+    amount,
+  });
+
+  return signWithKeypair(signingData, wallet.privateKey);
+}
+
+/**
+ * Verifies that the given transaction has a valid signature based on public-key encryption.
+ *
+ * @param tx - A transaction to verify the signature of. (Can be in object or encoded string format).
+ * @returns Returns true if tx has a valid signature, and returns false otherwise.
+ */
+function verify(tx: Transaction | string): boolean {
+  const decodedTx: Transaction = getDecodedTransaction(tx);
+  return verifySignature(
+    encodeForSigning(decodedTx),
+    decodedTx.TxnSignature,
+    decodedTx.SigningPubKey
+  );
+}
+
+/**
+ * Combine takes a collection of signedTransactions with the same underlying transaction and produces a
+ * Transaction with all Signers. It then signs that Transaction and gives an id based on the combined Transaction.
+ *
+ * @param signedTransactions - A collection of the same transaction signed by different signers. The only difference
+ * between the elements of signedTransactions should be the Signers field.
+ * @returns An object with the combined transaction (now having a sorted list of all signers) which is encoded, along
+ * with a transaction id based on the combined transaction.
+ */
+function combine(signedTransactions: string[]): SignedTransaction {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- decode gives a JsonObject which we know is a Transaction, so we cast it to get strong-typing in this code and our exported interface
+  const transactions: Transaction[] = signedTransactions.map((tx: string) => {
+    return decode(tx);
+  }) as unknown as Transaction[];
+
+  // This will throw a more clear error for JS users if there's a problem with any of the transaction's formatting
+  transactions.forEach((tx) =>
+    verifyBaseTransaction(tx as unknown as Record<string, unknown>)
```

Comment:
This should be replaced with `verify` later

Reply:
You shouldn't have to use type casting here.

Reply:
I'm not sure how to avoid it, I suppose an alternative would be to turn off the Typescript warning somehow? (Or maybe just have the inputs for Verify be Transactions, since those type descriptions only apply to TypeScript anyway?) - @mvadari What do you think would be best? (Or does Verify make this error irrelevant?)

Reply:
There's a lot of annoying linter errors to wade through if we type them as `Transaction`s, and I also don't think that's actually correct. 
I'm surprised that TypeScript gets mad at this issue, because a `Transaction` is technically a `Record<string, unknown`. That's one error that we can't just "turn off", though.